README.md
# BananaSlider

## To start:

```html
<body>
  <!-- If you just want a simple init, put this at the bottom of your "body" tag  -->
  <script src="https://banana.kruceo.com/dist/bundle.cdn.js"></script>
  <script>
    initAllSliders();
  </script>
</body>
```

## Your first step:

```html
<slider>
  <slide>
    <h2>Your first slide!</h2>
  </slide>
  <slide>
    <h2>Your second slide!</h2>
  </slide>
  <slide>
    <h2>Your third slide!</h2>
  </slide>
</slider>
```

## Customize showtime:

```html
<slider cooldown="5000">
  <!-- 5 seconds -->
  <slide>
    <h2>5 seconds to change bro!</h2>
  </slide>
  <slide>
    <h2>More 5 seconds!</h2>
  </slide>
  <slide>
    <h2>Wait a little bit...</h2>
  </slide>
</slider>
```

To a stopped slide:

```html
<slider cooldown="0">
  <!-- it doesn't move, other than with "hover:'add'" attribute or js script commands -->
  <slide>
    <h2>I don't move any muscles</h2>
  </slide>
  <slide>
    <h2>More a eternity of seconds!</h2>
  </slide>
  <slide>
    <h2>wait a lot...</h2>
  </slide>
</slider>
```

To individual slide showtime set:

```html
<slider>
  <!-- if you don't set a cooldown to "slider" tag, the default is 1 second-->
  <slide cooldown="7000">
    <h2>I'm here for 7 seconds</h2>
  </slide>
  <slide cooldown="5000">
    <h2>I'm here for 5 seconds</h2>
  </slide>
  <slide cooldown="3000">
    <h2>I'm here for 3 seconds</h2>
  </slide>
</slider>
```

## Customize animation speed:

```html
<slider speed="250">
  <!-- 0.25 seconds -->
  <slide>
    <h2>I'll complete this transition in 250 millis</h2>
  </slide>
  <slide>
    <h2>Is this fast?</h2>
  </slide>
  <slide>
    <h2>You like?</h2>
  </slide>
</slider>
```

## Customize hover action:

```html
<slider hover="add">
  <!-- "add", "stop" or "none"... "stop" is default -->
  <slide>
    <h2>Put your mouse over me!</h2>
  </slide>
  <slide>
    <h2>BOOOOOOOO!</h2>
  </slide>
</slider>
```

## Customize movement:

```html
<slider movement="vertical">
  <!-- "vertical" and "horizontal"... "horizontal" is default -->
  <slide>
    <h2>I'll move to the top!</h2>
  </slide>
  <slide>
    <h2>Shooting stars</h2>
  </slide>
  <slide>
    <h2>:)</h2>
  </slide>
</slider>
```

## Customize direction:

```html
<slider direction="backward">
  <!-- "forward" is default -->
  <slide>
    <h2>I'll move to the right!</h2>
  </slide>
  <slide>
    <h2>All right?</h2>
  </slide>
  <slide>
    <h2>With you?</h2>
  </slide>
</slider>
```
## Customize curve:

```html
<slider curve="cubic-bezier(1, 0, 0, 1)">
  <!-- Your animation style -->
  <slide>
    <h2>JS</h2>
  </slide>
  <slide>
    <h2>CSS</h2>
  </slide>
  <slide>
    <h2>HTML</h2>
  </slide>
</slider>
```

## Customize initial slide:

```html
<slider initial="1">
  <!-- Show 2023 first" -->
  <slide>
    <h2>2022</h2>
  </slide>
  <slide>
    <h2>2023</h2>
  </slide>
  <slide>
    <h2>2024</h2>
  </slide>
</slider>
```

## How to control this with JS?

All sliders are initialized on window load and indexed on "banana" object with your id attribute name, and you can find your slider calling the object "banana"...
<br/>

###### PS: If you don't give a id to your slide is set the default name and you can reach it like any other, the name is set like "slider1", "slider2"...

##### Write your slider like you prefer and give a id attribute...

```html
<slider id="my-slider">
  <!-- the obj name is "my-slider" -->
  <slide> 1 </slide>
  <slide> 2 </slide>
  <slide> 3 </slide>
  <slide> 4 </slide>
</slider>
```

##### Now you can control it calling `banana['my-slider']`

```html
<button onClick="banana['my-slider'].move(1)">+</button>
<button onClick="banana['my-slider'].move(-1)">-</button>
```

### What is move?

move adds the parameter number to current slider index, try this:

```html
<button onClick="banana['my-slider'].move(1)">+1</button>
<button onClick="banana['my-slider'].move(-1)">-1</button>
<button onClick="banana['my-slider'].move(2)">+2</button>
<button onClick="banana['my-slider'].move(-2)">-2</button>
```

### What is tp?

tp is acronym to "teleport" and do exactly this, a example:

```html
<button onClick="banana['my-slider'].tp(0)">1</button>
<button onClick="banana['my-slider'].tp(1)">2</button>
<button onClick="banana['my-slider'].tp(2)">3</button>
<button onClick="banana['my-slider'].tp(3)">4</button>
```
### What is getIndex?
the getIndex function return the current slide that's showing:

```js
setInterval(()=>
{
  let i = banana['my-slider'].getIndex();
  console.log(i)
},1000)
```
## Using the show event:

The most basic use of the event is with the attribute "onShow":

```html
<slider initial="1">
  <slide>
    <h2>I don't print anything</h2>
  </slide>
  <slide onShow="console.log('BANANA')">
    <h2>I print 'BANANA'</h2>
  </slide>
  <slide>
    <h2>I don't print anything</h2>
  </slide>
</slider>
```

Now set a id to any slide, like this:

```html
<slide id='slide1'>
    <h2>Print anything</h2>
</slide>
```
In your javascript file or tag, use:

```javascript
document.querySelector("#slide1").addEventListener('show',(e)=>
{
    console.log('Anything');
})
```

## More

You can see more and in real time working examples in <a href='https://banana.kruceo.com'>banana.kruceo.com


NEWS.md
# News

* getIndex() => documented 
* onShow event  => documented
* curveAttr => documented
* directtion attt ==> not documented

slide.js
let threads = [];
let animationSpeed = 500;
let cooldown = 200;
let hover = 'stop';
let animation = 'left';
let initial = 0;
export let banana = [];
function init() {
    banana = [];
    const sliderList = document.querySelectorAll("slider")
    if (sliderList.length > 0) {
        for (var slider of sliderList) {
            let thread = startSlider(slider).then((id)=>
            {
                console.log(id);
            })
            registerSlider(slider,thread)
        }
    }
    else {
        console.warn('[banana-slider] No sliders here!\nIf you need help try check banana.kruceo.com\nCase you don\'t like this message, add banana.mess = false');
    }
}

window.onload = function () {
   init();
}

async function startSlider(yourSlider) {

    let slider = yourSlider;
    if (slider.getAttribute('cooldown')) {
        cooldown = slider.getAttribute('cooldown');
    }
    else {
        cooldown = 2000;
    }
    if (slider.getAttribute('speed')) {
        animationSpeed = slider.getAttribute('speed');
    }
    else {
        animationSpeed = 500;
    }
    if (slider.getAttribute('hover')) {
        hover = slider.getAttribute('hover');
    }
    else {
        hover = 'stop';
    }
    if (slider.getAttribute('animation')) {
        animation = slider.getAttribute('animation');
    }
    else {
        animation = 'left';
    }
    if (slider.getAttribute('initial')) {
        initial = slider.getAttribute('initial');
    }
    else {
        initial = 0;
    }
    slider.innerHTML = '<slider-frame>' + slider.innerHTML + '</slider-frame>';
    let frame = slider.querySelector('slider-frame');
    let itens = frame.querySelectorAll('slide');
    let index = initial;
    let max = itens.length;

    slider.style.setProperty("overflow", "hidden");
    slider.style.setProperty("display", "block");
    
    switch (animation) {
        case 'left':
            frame.style = "position: relative;left: 0%;display: grid;grid-auto-flow: column;grid-auto-columns: 1fr;--total-items:" + (itens.length) + ";width: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;left: calc(var(--index)*-100%);transition: left var(--speed)";
            break;
        case 'right':
            frame.style = "position: relative;left: 0%;display: grid;grid-auto-flow: column;grid-auto-columns: 1fr;--total-items:" + (itens.length) + ";width: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;left: calc(var(--index)*100%);transition: left var(--speed)";
            break;
        case 'up':
            frame.style = "position: relative;top: 0%;--total-items:" + (itens.length) + ";height: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;top: calc(var(--index)*-100%);transition: top var(--speed)";
            break;

        default:
            break;
    }
    if (cooldown > 0) {
        threads[threads.length] = setInterval(function () {
            frame.style.setProperty('--index', index);
           
            index++;
            if (index > max - 1) {
                index = 0;
            }
        }, cooldown);
    }
    let currentIndex = 0;
    switch (hover) {
        case 'stop':

            slider.addEventListener('mouseover', function (event) {
                currentIndex = frame.style.getPropertyValue('--index')
                frame.style.setProperty('--speed', '100000s');
            }, true)
            slider.addEventListener('mouseout', function (event) {
                index = Number.parseInt(currentIndex);
                currentIndex = 0;
                frame.style.setProperty('--speed', animationSpeed + 'ms');
            }, true)
            break;
        case 'add':
            currentIndex = frame.style.getPropertyValue('--index')

            slider.addEventListener('mouseover', function (event) {
                frame.style.setProperty('--index', currentIndex + 1);
            }, true)
            slider.addEventListener('mouseout', function (event) {
                frame.style.setProperty('--index', currentIndex);
            }, true)
            break;

        default:
            break;
    }
    return threads[threads.length - 1]
}


function registerSlider(element,thread) {
    let name = element.getAttribute('id') || 'slider' + Object.keys(banana).length
    banana[name] = {
        el: element,
        thread: thread,
        name: name,
        tp: (newIndex) => { goToSlide(element, newIndex) },
        move: (value) => { moveSlider(element, value) }
    }
}

async function stopAllSlider() {
    threads.forEach(element => {
        clearInterval(element);
    });
}

async function moveSlider(yourSliderId, value) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    const max = Number.parseInt(slider.style.getPropertyValue('--total-items'))
    const index = Number.parseInt(slider.style.getPropertyValue('--index'))
    var addValue = value;
    if (!value) {
        addValue = 1;
    }
    if (index == 0 && value < 0) {
        addValue = 0;
    }
    if (index >= max - 1 && value > 0) {
        addValue = -(max - 1);
    }

    slider.style.setProperty('--index', index + addValue)
}
async function goToSlide(yourSliderId, value) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    const max = Number.parseInt(slider.style.getPropertyValue('--total-items'))
    //const index = Number.parseInt(slider.style.getPropertyValue('--index'))
    var addValue = value;
    if (!value) {
        addValue = 0;
    }
    if (addValue < 0) {
        addValue = 0;
    }
    if (addValue > max - 1) {
        addValue = max - 1;
    }

    slider.style.setProperty('--index', addValue)

}

async function deleteSlider(yourSliderId) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    //const index = Number.parseInt(slider.style.getPropertyValue('--index'))
    slider.style.setProperty('--index', addValue)

}


src/cdn.js
let threads = [];
let animationSpeed = 500;
let cooldown = 200;
let hover = 'stop';
let animation = 'left';
let initial = 0;
let banana = [];
function initAllSliders() {
    banana = [];
    const sliderList = document.querySelectorAll("slider")
    if (sliderList.length > 0) {
        for (var slider of sliderList) {
            let thread = startSlider(slider);
            registerSlider(slider,thread);
        }
    }
    else {
        console.warn('[banana-slider] No sliders here!\nIf you need help try check banana.kruceo.com\nCase you don\'t like this message, add banana.mess = false');
    }
}

async function startSlider(yourSlider) {

    let slider = yourSlider;
    if (slider.getAttribute('cooldown')) {
        cooldown = slider.getAttribute('cooldown');
    }
    else {
        cooldown = 2000;
    }
    if (slider.getAttribute('speed')) {
        animationSpeed = slider.getAttribute('speed');
    }
    else {
        animationSpeed = 500;
    }
    if (slider.getAttribute('hover')) {
        hover = slider.getAttribute('hover');
    }
    else {
        hover = 'stop';
    }
    if (slider.getAttribute('animation')) {
        animation = slider.getAttribute('animation');
    }
    else {
        animation = 'left';
    }
    if (slider.getAttribute('initial')) {
        initial = slider.getAttribute('initial');
    }
    else {
        initial = 0;
    }
    slider.innerHTML = '<slider-frame>' + slider.innerHTML + '</slider-frame>';
    let frame = slider.querySelector('slider-frame');
    let itens = frame.querySelectorAll('slide');
    let index = initial;
    let max = itens.length;

    slider.style.setProperty("overflow", "hidden");
    slider.style.setProperty("display", "block");
    
    switch (animation) {
        case 'left':
            frame.style = "position: relative;left: 0%;display: grid;grid-auto-flow: column;grid-auto-columns: 1fr;--total-items:" + (itens.length) + ";width: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;left: calc(var(--index)*-100%);transition: left var(--speed)";
            break;
        case 'right':
            frame.style = "position: relative;left: 0%;display: grid;grid-auto-flow: column;grid-auto-columns: 1fr;--total-items:" + (itens.length) + ";width: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;left: calc(var(--index)*100%);transition: left var(--speed)";
            break;
        case 'up':
            frame.style = "position: relative;top: 0%;--total-items:" + (itens.length) + ";height: calc(var(--total-items) * 100%);--index: 0;--speed: " + animationSpeed + "ms;top: calc(var(--index)*-100%);transition: top var(--speed)";
            break;

        default:
            break;
    }
    if (cooldown > 0) {
        threads[threads.length] = setInterval(function () {
            frame.style.setProperty('--index', index);
           
            index++;
            if (index > max - 1) {
                index = 0;
            }
        }, cooldown);
    }
    let currentIndex = 0;
    switch (hover) {
        case 'stop':

            slider.addEventListener('mouseover', function (event) {
                currentIndex = frame.style.getPropertyValue('--index')
                frame.style.setProperty('--speed', '100000s');
            }, true)
            slider.addEventListener('mouseout', function (event) {
                index = Number.parseInt(currentIndex);
                currentIndex = 0;
                frame.style.setProperty('--speed', animationSpeed + 'ms');
            }, true)
            break;
        case 'add':
            currentIndex = frame.style.getPropertyValue('--index')

            slider.addEventListener('mouseover', function (event) {
                frame.style.setProperty('--index', currentIndex + 1);
            }, true)
            slider.addEventListener('mouseout', function (event) {
                frame.style.setProperty('--index', currentIndex);
            }, true)
            break;

        default:
            break;
    }
    return threads[threads.length - 1]
}


function registerSlider(element,thread) {
    let name = element.getAttribute('id') || 'slider' + Object.keys(banana).length
    banana[name] = {
        el: element,
        thread: thread,
        name: name,
        tp: (newIndex) => { goToSlide(element, newIndex) },
        move: (value) => { moveSlider(element, value) }
    }
}

async function stopAllSlider() {
    threads.forEach(element => {
        clearInterval(element);
    });
}

async function moveSlider(yourSliderId, value) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    const max = Number.parseInt(slider.style.getPropertyValue('--total-items'))
    const index = Number.parseInt(slider.style.getPropertyValue('--index'))
    var addValue = value;
    if (!value) {
        addValue = 1;
    }
    if (index == 0 && value < 0) {
        addValue = 0;
    }
    if (index >= max - 1 && value > 0) {
        addValue = -(max - 1);
    }

    slider.style.setProperty('--index', index + addValue)
}
async function goToSlide(yourSliderId, value) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    const max = Number.parseInt(slider.style.getPropertyValue('--total-items'))
    //const index = Number.parseInt(slider.style.getPropertyValue('--index'))
    var addValue = value;
    if (!value) {
        addValue = 0;
    }
    if (addValue < 0) {
        addValue = 0;
    }
    if (addValue > max - 1) {
        addValue = max - 1;
    }

    slider.style.setProperty('--index', addValue)

}

async function deleteSlider(yourSliderId) {
    const slider = yourSliderId.getElementsByTagName('slider-frame')[0];
    slider.style.setProperty('--index', addValue);

}


rollup.config.js
import fs from "fs";
import cp from 'child_process';
const browserify = (target) => {
  return {
    name: "browserify",
    renderChunk(code, chunk, options) {
      if (options.format != "es") return;

      setTimeout(() => {
        let file = code.toString();
        let lines = file.split("\n").slice(0, -1);
        file = lines.reduce((p, c) => p + c + "\n", "");
        fs.writeFileSync(target.file, file);
      }, 100);
    },
  };
};
const genTsMod = (target) => {
  return {
    name: "genTsMod",
    renderChunk(code, chunk, options) {
      if (options.format != "cjs") return;

      setTimeout(() => {
       cp.exec('npx -p typescript tsc dist/bundle.cjs.js --declaration --allowJs --emitDeclarationOnly --outDir dist')
      }, 100);
    },
  };
};
const minify = (target) => {
  return {
    name: "minify",
    renderChunk(code, chunk, options) {
      setTimeout(() => {
        let file = code.toString();
        file = file.replaceAll('\n','').replaceAll('  ','');
        fs.writeFileSync(target.file.replace('@@@',options.format),file)
      }, 100);
    },
  };
};
export default {
  input: "src/index.js",
  output: [
    {
      file: "dist/bundle.cjs.js",
      format: "cjs",
    },
    {
      file: "dist/bundle.es.js",
      format: "es",
    },
    
  ],
  plugins: [browserify({ file: "dist/bundle.cdn.js" }),genTsMod({ file: "dist/bundle.cjs.d.ts" }),minify({file: "dist/bundle.min.@@@.js"})],
};


src/index.js
let threads = [];
let animationSpeed = 500;
let cooldown = 200;
let hover = "stop";
let movement = "left";
let initial = 0;
let banana = [];
let animationCurve = "";
let direction = "forward";
const evt = new Event("show");
function initAllSliders() {
  banana = [];
  const sliderList = document.querySelectorAll("slider");
  if (sliderList.length > 0) {
    for (var slider of sliderList) {
      let thread = startSlider(slider);
      registerSlider(slider, thread);
    }
  } else {
    console.warn(
      "[banana-slider] No sliders here!\nMay you need help try check banana.kruceo.com\nCase you don't like this message, add banana.mess = false"
    );
  }
}

async function startSlider(yourSlider) {
  let slider = yourSlider;
  cooldown = slider.getAttribute("cooldown") ?? 2000;
  animationSpeed = slider.getAttribute("speed") ?? 500;
  hover = slider.getAttribute("hover") ?? "stop";
  movement = slider.getAttribute("movement") ?? "horizontal";
  initial = slider.getAttribute("initial") ?? 0;
  animationCurve = slider.getAttribute("curve")??'cubic-bezier(0.075, 0.82, 0.165, 1)'
  direction = slider.getAttribute("direction")??"forward"
  if (!slider.querySelector("slider-frame")) {
    slider.innerHTML = "<slider-frame>" + slider.innerHTML + "</slider-frame>";
  }
  let frame = slider.querySelector("slider-frame");
  let itens = [...frame.querySelectorAll("slide")];
  itens.forEach((item) => {
    if (item.getAttribute("onShow") != null) {
      item.addEventListener("show", (e) => {
        try {
          let onSHowFunction = new Function(item.getAttribute("onShow"));
          onSHowFunction();
        } catch (error) {
          console.error(
            "[banana-slider] " +
              e.target.localName +
              " - " +
              e.target.id +
              " Slider onShow attribute error \n" +
              error
          );
        }
      });
    }
  });
  let index = initial;
  let max = itens.length;

  slider.style.setProperty("overflow", "hidden");
  slider.style.setProperty("display", "block");

  switch (movement) {
    case "horizontal":
      frame.style =
        "position: relative;left: 0%;display: grid;grid-auto-flow: column;grid-auto-columns: 1fr;--total-items:" +
        itens.length +
        ";width: calc(var(--total-items) * 100%);--index: " +
        initial +
        ";--speed: " +
        animationSpeed +
        "ms;left: calc(var(--index)*-100%);transition: left var(--speed) "+animationCurve;
      break;
    case "vertical":
      slider.style.setProperty("height", itens.sort((a,b)=>a.offsetHeight - b.offsetHeight)[itens.length - 1].offsetHeight *1.5 + 'px' );
      itens.forEach((item)=>item.style.setProperty("height", itens.sort((a,b)=>a.offsetHeight - b.offsetHeight)[itens.length - 1].offsetHeight*1.5 + 'px' ));
      frame.style =
        "position: relative;top: 0%;display: flex;flex-direction: column;grid-auto-row: 1fr;--total-items:" +
        itens.length +
        ";--coef: 0"+
        ";height: calc(var(--total-items) * 100%)"+
        ";--index: " + initial +
        ";--speed: " + animationSpeed +
        ";ms;top: calc(var(--index)*-100%)"+
        ";transition: top var(--speed) "+animationCurve;
      break;
    default:
      break;
  }
  if (cooldown > 0) {
    threads[threads.length] = setLoop(() => {
      threads[threads.length - 1].delay = itens[index].getAttribute("cooldown") ?? cooldown;
      itens[index].dispatchEvent(evt);
      frame.style.setProperty("--index", index);
      if(direction == 'backward')index--;
      else{index++}
      if (index > max - 1) {
        index = 0;
      }if (index > max - 1) {
        index = 0;
      }
      if(index < 0)
      {
        index = max - 1;
      }
    }, itens[index].getAttribute("cooldown") ?? cooldown);
  }
  let currentIndex = 0;
  switch (hover) {
    case "stop":
      slider.addEventListener("mouseover",
        function (event) {
          currentIndex = frame.style.getPropertyValue("--index");
          frame.style.setProperty("--speed", "100000s");
        },
        true
      );
      slider.addEventListener("mouseout",
        function (event) {
          index = Number.parseInt(currentIndex);
          currentIndex = 0;
          frame.style.setProperty("--speed", animationSpeed + "ms");
        },
        true
      );
      break;
    case "add":
      currentIndex = frame.style.getPropertyValue("--index");

      slider.addEventListener(
        "mouseover",
        function (event) {
          frame.style.setProperty("--index", currentIndex + 1);
        },
        true
      );
      slider.addEventListener(
        "mouseout",
        function (event) {
          frame.style.setProperty("--index", currentIndex);
        },
        true
      );
      break;

    default:
      break;
  }
  return threads[threads.length - 1];
}

function registerSlider(element, thread) {
  let name =
    element.getAttribute("id") || "slider" + Object.keys(banana).length;
  banana[name] = {
    el: element,
    thread: thread,
    name: name,
    tp: (newIndex) => {
      goToSlide(element, newIndex);
    },
    move: (value) => {
      moveSlider(element, value);
    },
    getIndex: () => {
      return parseInt(
        element.querySelector("slider-frame").style.getPropertyValue("--index")
      );
    },
  };
}

async function stopAllSlider() {
  threads.forEach((element) => {
    clearInterval(element);
  });
}

async function moveSlider(yourSliderId, value) {
  const slider = yourSliderId.getElementsByTagName("slider-frame")[0];
  const max = Number.parseInt(slider.style.getPropertyValue("--total-items"));
  const index = Number.parseInt(slider.style.getPropertyValue("--index"));
  var addValue = value;
  if (!value) {
    addValue = 1;
  }
  if (index == 0 && value < 0) {
    addValue = 0;
  }
  if (index >= max - 1 && value > 0) {
    addValue = -(max - 1);
  }

  slider.style.setProperty("--index", index + addValue);
}
async function goToSlide(yourSliderId, value) {
  const slider = yourSliderId.getElementsByTagName("slider-frame")[0];
  const max = Number.parseInt(slider.style.getPropertyValue("--total-items"));
  var addValue = value;
  if (!value) {
    addValue = 0;
  }
  if (addValue < 0) {
    addValue = 0;
  }
  if (addValue > max - 1) {
    addValue = max - 1;
  }

  slider.style.setProperty("--index", addValue);
}

async function deleteSlider(yourSliderId) {
  const slider = yourSliderId.getElementsByTagName("slider-frame")[0];
  slider.style.setProperty("--index", addValue);
}
export function setLoop(func, initialDelay) {
  let thisLoop = {
    delay: initialDelay,
    start: () => {
      setTimeout(() => {
        if (!thisLoop.stopped) {
          func();
          thisLoop.start(thisLoop.delay);
          return;
        }
      }, thisLoop.delay);
    },
    stopped: false,
    stop: () => {
      thisLoop.stopped = true;
    },
  };
  thisLoop.start();
  return thisLoop;
}
export { banana, initAllSliders };
